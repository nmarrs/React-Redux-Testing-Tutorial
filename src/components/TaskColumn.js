import React from 'react'
import PropTypes from 'prop-types'

import './styles/TaskColumn.css'
import TaskCard from './TaskCard'

const TaskColumn = (props) => {
  const {
    title,
    toggleEditTaskModal,
    tasks
  } = props

  return (
    <div className="card shadow font-title mt-3 mb-5 TaskColumn-card mx-auto">
      <div className='card-title mt-3'>
        <h3>{title} - {tasks ? tasks.length : 0}</h3>
      </div>
      <div className='card-body'>
        {
          tasks && tasks.length > 0
          ? Object.entries(tasks).map(([key, task]) => (
            <div key={key}>
                <TaskCard
                  task={task}
                  toggleEditTaskModal={toggleEditTaskModal} />
            </div>
          ))
          : <div>No tasks</div>
        }
      </div>
    </div>
  )
}

TaskColumn.propTypes = {
  title: PropTypes.string.isRequired,
  toggleEditTaskModal: PropTypes.func.isRequired,
  tasks: PropTypes.array
}

export default TaskColumn
