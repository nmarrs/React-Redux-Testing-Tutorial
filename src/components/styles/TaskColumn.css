.TaskColumn-card {
  min-width: 325px;
  text-align: center;
}
