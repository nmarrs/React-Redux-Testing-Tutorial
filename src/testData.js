export let testTask = {
  id: 1,
  title: 'Unload merchandise',
  priority: 3,
  timeEstimate: 2,
  notes: 'Unload merchandise from the bay door. Driver usually comes by at 6 am sharp.',
  feedback: '',
  startTime: '',
  endTime: '',
  completionTime: '',
  currentStatus: 'To Do',
  rankWeight: 150
}

export let testTask1 = {
  id: 2,
  title: 'Stock shelves with unpacked items',
  priority: 1,
  timeEstimate: 10,
  notes: 'Take merchandise items and stock shelves with them. Be sure to place items in correct areas.',
  feedback: '',
  startTime: '',
  endTime: '',
  completionTime: '',
  currentStatus: 'To Do',
  rankWeight: 10
}

export let testTask2 = {
  id: 3,
  title: 'Stock shelves with unpacked items',
  priority: 5,
  timeEstimate: 2,
  notes: 'Take merchandise items and stock shelves with them. Be sure to place items in correct areas.',
  feedback: '',
  startTime: '',
  endTime: '',
  completionTime: '',
  currentStatus: 'To Do',
  rankWeight: 250
}
