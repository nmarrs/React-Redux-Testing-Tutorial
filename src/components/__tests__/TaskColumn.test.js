import React from 'react'
import { shallow } from 'enzyme'

import TaskColumn from '.././TaskColumn'
import { testTask, testTask1, testTask2 } from '../../testData'

describe('TaskColumn', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<TaskColumn
      title={'Test'}
      toggleEditTaskModal={jest.fn()}
      tasks={[]} />)
  })

  it('should exist', () => {
    expect(wrapper).toBeDefined()
  })

  describe('renders correctly', () => {
    it('no tasks provided', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('tasks provided', () => {
      wrapper.setProps({tasks: [testTask, testTask1, testTask2]})
      expect(wrapper).toMatchSnapshot()
    })
  })
})
