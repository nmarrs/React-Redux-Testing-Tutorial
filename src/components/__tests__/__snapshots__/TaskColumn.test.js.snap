// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskColumn renders correctly no tasks provided 1`] = `
<div
  className="card shadow font-title mt-3 mb-5 TaskColumn-card mx-auto"
>
  <div
    className="card-title mt-3"
  >
    <h3>
      Test
       - 
      0
    </h3>
  </div>
  <div
    className="card-body"
  >
    <div>
      No tasks
    </div>
  </div>
</div>
`;

exports[`TaskColumn renders correctly tasks provided 1`] = `
<div
  className="card shadow font-title mt-3 mb-5 TaskColumn-card mx-auto"
>
  <div
    className="card-title mt-3"
  >
    <h3>
      Test
       - 
      3
    </h3>
  </div>
  <div
    className="card-body"
  >
    <div
      key="0"
    >
      <TaskCard
        task={
          Object {
            "completionTime": "",
            "currentStatus": "To Do",
            "endTime": "",
            "feedback": "",
            "id": 1,
            "notes": "Unload merchandise from the bay door. Driver usually comes by at 6 am sharp.",
            "priority": 3,
            "rankWeight": 150,
            "startTime": "",
            "timeEstimate": 2,
            "title": "Unload merchandise",
          }
        }
        toggleEditTaskModal={[MockFunction]}
      />
    </div>
    <div
      key="1"
    >
      <TaskCard
        task={
          Object {
            "completionTime": "",
            "currentStatus": "To Do",
            "endTime": "",
            "feedback": "",
            "id": 2,
            "notes": "Take merchandise items and stock shelves with them. Be sure to place items in correct areas.",
            "priority": 1,
            "rankWeight": 10,
            "startTime": "",
            "timeEstimate": 10,
            "title": "Stock shelves with unpacked items",
          }
        }
        toggleEditTaskModal={[MockFunction]}
      />
    </div>
    <div
      key="2"
    >
      <TaskCard
        task={
          Object {
            "completionTime": "",
            "currentStatus": "To Do",
            "endTime": "",
            "feedback": "",
            "id": 3,
            "notes": "Take merchandise items and stock shelves with them. Be sure to place items in correct areas.",
            "priority": 5,
            "rankWeight": 250,
            "startTime": "",
            "timeEstimate": 2,
            "title": "Stock shelves with unpacked items",
          }
        }
        toggleEditTaskModal={[MockFunction]}
      />
    </div>
  </div>
</div>
`;
